### DISCUSSION TITLE
`case` on the same line as `:` in partial functions

### DISCUSSION THREAD

**Discussion by joan38** [scala/scala3#22016] [Feature Requests]
Hi,

Any reasons why not to make the following compile:
```scala
elements.collect: case s: String =>
  s
```
```
[error] an identifier expected, but 'case' found
[error] elements.collect: case s: String =>
[error]                   ^^^^
```
https://scastie.scala-lang.org/asibTVDBSWiaMPZEU1I9nA

as opposed to the compiling:
```scala
elements.collect:
  case s: String => s
```
https://scastie.scala-lang.org/C9LRPacdQsib1wMFdQUKYg

Especially that the following works:
```scala
elements.map: s =>
  s
```
https://scastie.scala-lang.org/mHxCBmcQSS6lVLXHto24tw

Thanks

**Comment by som-snytt** [comment]
In this neighborhood
```
scala> try ??? catch case t => 42
val res0: Int = 42

scala> 27 match case t => 42
-- [E040] Syntax Error: ------------------------------------------------------------------------------------------------
1 |27 match case t => 42
  |         ^^^^
  |         '{' expected, but 'case' found
```

**Comment by joan38** [comment]
I'd be happy to contribute this but I have no idea where to start. Can someone hint at where to look to implement such change?

**Comment by som-snytt** [comment]
"Parens vs braces" is the new "tabs vs spaces".

I expressed hope that the new SIP to allow parens around cases will also cover the "one-liner" examples.

The motivation is that if it allows the one-liner
```
xs.map(case t: T => f(t))
```
then it's also reasonable to accept
```
xs.map: case t: T => f(t)
```
and then when it's not a one-liner but still a single case
```
xs.map: case t: T =>
  val v = f(t)
  v
```
which is further motivated by manual parameter untupling
```
pairs.map: case (x, y) =>
  body
```
Note that it does not yet accept one-liners after colon:
```
xs.map: x => x + 1
```
The previous judgment was that braceless colon is not intended to eliminate parens; but maybe that will change with the new front in the war between braces and parens. That is, all the options for bracketing code get taken to their logical conclusion in the brackocalypse. Abrackalypse?

https://contributors.scala-lang.org/t/sip-xx-allow-partial-function-literals-to-be-defined-with-parentheses/7207/12?u=som-snytt

**Comment by som-snytt** [comment]
`-language:experimental.relaxedLambdaSyntax`

**Reply by joan38** [reply]
Will this be available in the 3.7.x series? 
We are stuck on 3.7 until Spark moves off 2.13.

**Reply by som-snytt** [reply]
I assume no more 3.7.x because it is not LTS. Fixes are backported to 3.3 LTS but probably not experimental features. 3.9 is next LTS? IIRC. At least this is just convenience syntax.

**Reply by SethTisue** [reply]
yeah 3.7.5 will almost certainly never happen (and if it did happen, it would be because some critical fix was needed, not to open the door for random backports, especially new syntax)
